src/noise.rs

```diff
@@ -47,7 +51,9 @@ pub fn gaussian_white_noise(s: Signal, signal_to_noise: f64, sample_duration: Ti
     let rms_signal = rms(&s, sample_duration);
     let rms_noise_squared = rms_signal.powf(2.0) / (10.0f64.powf(signal_to_noise / 10.0));
     let dist = Normal::new(0.0, rms_noise_squared);
-    let noise = move |_| dist.sample(&mut rand::thread_rng());
+    let salt: f64 = SmallRng::from_entropy().gen();
+    let noise =
+        move |t: TimeSecs| dist.sample(&mut SmallRng::seed_from_u64((t.0 * 1000.0 + salt) as u64));
```

Comment:
nice.
I looked at the implementation, and making a SmallRng doesn't seem to do very much computation or use much memory. So we can make a lot of noise, secure in knowing what we are seeing is what we are hearing and that the wave composition is truth.